src/WarcraftLegacies.Source/Quests/Warsong/QuestScepterOfTheQueenWarsong.cs

```diff
@@ -0,0 +1,35 @@
+﻿using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.QuestSystem;
+using MacroTools.QuestSystem.UtilityStructs;
+using WarcraftLegacies.Source.Setup;
+using WarcraftLegacies.Source.Setup.Legends;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace WarcraftLegacies.Source.Quests.Warsong
+{
+    public class QuestScepterOfTheQueenWarsong : QuestData
+    {
+        public QuestScepterOfTheQueenWarsong(Rectangle area) : base("Royal Plunder", "Remnants of the ancient Highborne survive within the ruins of Dire Maul. If Feathermoon Stronghold falls, it would become a simple matter to slaughter the Highborne and plunder their artifacts.", "ReplaceableTextures\\CommandButtons\\BTNNagaWeaponUp2.blp")
+        {
+            HighBourneArea = area;
+            AddObjective(new ObjectiveLegendNotPermanentlyDead(LegendWarsong.StonemaulKeep));
+            AddObjective(new ObjectiveLegendDead(LegendSentinels.legendFeathermoon));
+            AddObjective(new ObjectiveAnyUnitInRect(HighBourneArea, "Dire Maul", true));
+            HighBourneArea.Rect.MakeUnitsInvulnerable();
+        }
+
+        private Rectangle HighBourneArea { get; }
+        protected override string CompletionPopup => "The Highborne are no longer implicitly defended by the Night Elven presence at Feathermoon Stronghold. The Horde unleashes their full might against these Night Elven arcanists.";
+        protected override string RewardDescription => "Gain the Scepter of the Queen and turn all units in Dire Maul hostile";
+
+        protected override void OnComplete(Faction whichFaction)
+        {
+            base.OnComplete(whichFaction);
```

Comment:
Do you think we need to call the base? If we do I might need to add this to every other quest.

Reply:
No the call is not neccessary since the base method is not doing anything. I removed it.
If there are no cases where the virtual methods in QuestData have default behaviour they could all be made abstract to enforce overrides in child classes

Reply:
Cool, makes sense. I can guarantee the base methods will never do anything.
At the same time I don't think we want to make them `abstract` since we don't really want to force behaviour override. E.g. there are many quests which don't do anything on failure, so why enforce writing something for that method? Up for debate.